.github/CONTRIBUTING.md

```diff
@@ -1,20 +1,46 @@
 
-Any typo fixes, spelling fixes, fixes in documentation,
-or README.md can be done without any further considerations!
-Simply do the fixes in your cloned repo and create a pull request.
+## Configure Git
 
-# Issues
+```sh
+$ git config --global user.name "Your Name"
+$ git config --global user.email "your_email@example.com"
+```
 
-If you have any questions you can ask them using
-[github issue](https://github.com/ElektraInitiative/libelektra/issues/new).
-Simply ignore the template in this case.
+When you connect to a GitHub repository from Git, you have to set up the [authentication](https://help.github.com/en/articles/set-up-git#next-steps-authenticating-with-github-from-git) settings.
 
-See also [FAQ](/doc/help/elektra-faq.md) for some details
-about creating issues.
+## Fork & Sync
+
+We use a fork/sync and pull-request (see below) model at GitHub, follow this short [tutorial](https://help.github.com/articles/fork-a-repo/) to get familiar with forking and syncing.
+
+Sync the fork of the repository to keep it up-to-date with the upstream repository by using the following git commands in your local repository. Important: Using these commands will avoid unnecessary merge commits while you are working on a pull-request (see below):
+
+```sh
+$ git fetch upstream
+$ git rebase upstream/master
+$ git push origin master --force
+```
+
+## Issues
+
+Check the [Ideas](/doc/IDEAS.md) page if you are searching for a good topic to start with. You can also visit the [issue-tracker](https://github.com/ElektraInitiative/libelektra/issues). Do not hesitate to open a new issue if you want to ask a question, report or fix a bug or approach new topics.
+
+## Creating a Pull-Request
+
+If you want to publish your local changes to this project you have to create a new pull-request.
+
+1. Open GitHub and navigate to _your_ libeketra-fork
+2. In the tab _Codes_ press the _New Pull-Request_ button and choose a title and fill in the [pull-request template](/.github/PULL_REQUEST_TEMPLATE.md)
+3. Now you should be able to set up the _Comparing changes_ settings. Choose the head repository and base repository. E.g.: if you have some changes in the master branch of your forked repository, select *head repository: yournickname/libelektra and *compare: master* and *base repositroy: ElektraInitiative/libelektra* and *base: master\*
+4. Add some information about the changes in the release notes (path of the file: /libelektra/doc/news/\_preparation_next_release.md), skipping this step may cause a rejected pull-request
+5. Commit and push your local changes in git (keep in mind to sync your fork - fetch, rebase & push)
+6. Wait for the code-review
+
+## Code-Review
+
+After the pull-request your code will be reviewed and if your pull-request passes the review , your changes will be merged to master.
```

Comment:
You could also write that in many cases the build fails without the fault of the developer. (e.g., websites are down, build server stuck (as you can see yourself)). Some short section about these issues would be helpful. Also a link to the buildserver is helpful: doc\BUILDSERVER.md

Reply:
Yes, we really should mention that in the docu, the build server problems are too often to not say it.